### ISSUE TITLE
Add consistent Units information in the json field tables

### ISSUE THREAD

**Issue by guiomar** [bids-standard/bids-specification#663]
Hi!

I think it would be useful to include consistent information for units in the table descriptions for fields in the json files.
 

1.  They could either go in new column, but maybe it's not worth the extra space, as it doesn't apply to every row.

1. Alternatively, they could go in the datatype column after the datatype in brackets, as they mainly apply to numerical data, for example, `SamplingFrequency`, numerical **[Hz]**. 

1.  Or other option could be at the beginning of the description column, being always consistent, for example: `SamplingFrequency`, numerical, **[Hz]** Sampling frequency of all the data in the recording, regardless of their type (for example, 2400).

I would vote for 2 or 3

**Comment by sappelhoff** [comment]
Do you have a particular example or use-case where you have looked at the specification and wondered for a field, what units they were in? (and how did you find out when you didn't see a direct indication in the datatype column or in a dedicated unit column?)

**Comment by satra** [comment]
in the nidm-terms project we are adding this to the json description of the column headers.  (cc: @dbkeator )

**Comment by guiomar** [comment]
In general I find the units in the description field, I only have to read a bit. And when I don't I assume is in SI, as it's recommended. Can't recall now. When reviewing the new beps, I try to ensure it's always indicated. However, I think it can be useful that this info (which is quite important), could be found straightforward at a glance, not having to read several lines to find it. This can just be at the beginning of the description in brackets or similar.

**Comment by sappelhoff** [comment]
> However, I think it can be useful that this info (which is quite important), could be found straightforward at a glance, not having to read several lines to find it.

True, but there are many JSON fields that do not have units, or where it's complicated, for example:

- `TaskName`, which is a string
- `SoftwareFilters`, which is an object of objects
- `FiducialsCoordinates`, which is an object of arrays ... AND the units for this are indicated in `FiducialsCoordinateUnits`

I feel like adding a "Units" column to these tables might introduce more clutter than clarity. WDYT @guiomar ?

For the NIDM Terms that's not a problem, because the terms are not displayed in a single table.

**Comment by tsalo** [comment]
I don't know if this helps, but adding Units (when units are specified in the schema) should be trivial with #774. Alternatively, we could try appending unit information and other info that's accessible from the schema (e.g., valid options when the field is an enum) into the Description column. That way that information doesn't actually have to go into the description field of the item's YAML file.

**Comment by satra** [comment]
i like @guiomar suggestion 2, but only for TSVs (this is convenient for users looking at the TSV). and this can be augmented with the json description for TSVs. for bids metadata i think it should go into the schema/term files since users can't assume their own units for those. but there is a lot of flexibility in the TSV space and having some consistency there would be useful.
